doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst

```diff
@@ -0,0 +1,303 @@
+It is used in :cpp:func:`default_signal_handler()` by default.
+
+If the network is left by a router or an end device, the device will try to join any open network.
+
+* The router will use the default signal handler to try to join or rejoin the network until it succeeds.
+* The end device will use the default signal handler to try to join or rejoin the network for a finite period of time, because the end devices are often powered by batteries.
+
+  * The procedure to join or rejoin the network is restarted after the device reset or power cycle.
+  * The procedure to join or rejoin the network can be restarted by calling :cpp:func:`user_input_indicate()`, but it needs to be implemented in the application (for example, by calling :cpp:func:`user_input_indicate()` when a button is pressed).
+    The procedure will be restarted only if the device does not join and the procedure is not running.
+
+The Zigbee rejoin procedure retries to join a network with each try after a specified amount of time: ``2^n`` seconds, where ``n`` is the number of retries.
+
+The period is limited to 15 minutes if the result is higher than that.
+
+* When :cpp:func:`start_network_rejoin()` is called, the rejoin procedure is started, and depending on the device role:
+
+  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
+    Once called, the alarm stops the rejoin.
+
+* When ``stop_network_rejoin(was_scheduled)`` is called, the network rejoin is canceled and the alarms scheduled by :cpp:func:`start_network_rejoin()` are canceled.
+
+  * Additionally for the end device, if :cpp:func:`stop_network_rejoin()` is called with ``was_scheduled`` set to ``ZB_TRUE``, :cpp:func:`user_input_indicate()` can restart the rejoin procedure.
+
+* For end devices only, :cpp:func:`user_input_indicate()` restarts the rejoin procedure if the device did not join the network and is not trying to join a network.
+  It is safe to call this function from an interrupt and to call it multiple times.
+
+
+.. warning::
+    The Zigbee network rejoin procedure is managed from multiple signals in :cpp:func:`default_signal_handler()`.
+    If the application controls the network joining, rejoining, or leaving, each signal in which the Zigbee network rejoin procedure is managed should be handled in the application.
+    In this case, :cpp:func:`user_input_indicate()` must not be called.
+
+.. _zarco_sleep:
+
+Zigbee stack sleep routines
+***************************
+
+The Zigbee stack provides the following mechanisms to reduce power consumption:
+
+* On Zigbee end devices, routers and coordinators:
+
+  * The stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`) if the Zigbee scheduler queue is empty.
+
+* On Zigbee sleepy end devices:
```

Comment:
To be simplified. Applies to all devices, starting from `If there are no delayed callbacks...`, with one exception: routers and coordinators do not turn off their timer.